### ISSUE TITLE
Complete unsubscribe page release (Epic)

### ISSUE THREAD

**Issue by rachelhbd** [350org/ak_intl_v3#332]
All related tickets are here: https://github.com/orgs/350org/projects/8/views/5

About 6 months ago we worked up a version of new unsubscribe page flow, but didn't have capacity to finish the project. Campaigners are keen to have this in place to get better data to understand our unsubscribe causes!

The code is in this branch, which is now quite far behind master & develop: https://github.com/350org/ak_intl_v3/tree/fa-nov-sprint-unsubscribe-page
but the relevant code is mostly in unsubscribe.html and thanks_unsubscribe.html which haven't been edited recently otherwise

Most recent GL ticket: https://gitlab.com/350/product-fa-sprints/-/issues/90 with a comment on the code from Jamie that might be helpful: 
> On the thank you page, we check if the page type is 'Unsubscribe'. If it is, we load in an extra template 'thanks_unsubscribe.html'. Otherwise, we just include the code for the original thank you page.
> https://github.com/350org/ak_intl_v3/blob/fa-nov-sprint-unsubscribe-page/thanks.html#L6

This feature includes #336 #337 #338 #339

Previews of current unsub page using FA templateset suggest that merging into master won't have any breaking changes for current unsub page: 
- [Recognised user, subscribed to 1 list](https://act.350.org/unsubscribe/unsubscribe/?akid=.3199019.2YIX0L&template_set=350.org%20-%20International%20v3%20(FA%20sprint%20nov%2021)):
- [Recognised user, subscribed to 2 lists](https://act.350.org/unsubscribe/unsubscribe?akid=.4350316.DJLspN&template_set=350.org%20-%20International%20v3%20(FA%20sprint%20nov%2021))
- the thanks page is redirected externally for now to: http://350.org/unsubscribed/ 

From previewing, existing unsub pages look functional using the new templateset except the user gets redirected so sees the thanks survey twice: 
https://act.350.org/unsubscribe/unsubscribe/?template_set=350.org%20-%20International%20v3%20FA%20Unsubscribe%20Page&akid=.5180668.GR9oqK

Release sequence:

- [x] Feature tweaks and fixes that need dev incl:
#336
#337
#339
#340 
#344 
#345
#343 
- [x] Make sure we have a plan for data tracking and analysis #348
- [x] Request translations
- [x] Merge into FA templateset
- [x] Complete tweaks just need edits in admin and translations #338
- [x] Last fix re dynamic name snippets for translations
- [x] Switch out the titles with dynamic names: {% if user %}, {{ user.first_name|default:"friend" }}{% endif %}
- [x] Update the page shortnames so they match those used in the templateset for the fallback url (where possible)
https://act.350.org/admin/core/multilingualcampaign/289/overview/
https://docs.google.com/spreadsheets/d/1LNa5Jnani1g_1YmJDFCHzVj2esJND9NadmPxQL3Z3Ss/edit#gid=0
- [x] Final QA & check translations (in Unsub template or Develop)
- [x] Use custom page redirect to redirect users from old to new pages whilst keeping the original urls for now. See redirected pages: https://act.350.org/report/7015/#savedQuery 
- [x] Create draft report until we have a dashboard to collate results: https://act.350.org/report/rh_new_unsubscribe_page_survey_responses_beta/
- [x] Merge templateset into Main/v3 when ready
- [x] update new thanks pages to use Main/v3: https://act.350.org/report/rachel_which_templatesets_are_new_unsubscibe_pages_using/
- [x] For EN, ES, FR, DE, PT update old & new slugs, so new pages don't have "new" in the slug and fallback urls are correct
- [x] For JA, BN, ID, NL, TR, ZH, update links in wrappers
- [x] QA pass of links in mailing previews and proofs https://act.350.org/mailings/drafts/compose/219925/
-> I caught that we needed to update the default unsubscribe page for any wrappers that used the default wrapper snippets, and that some multilingual wrappers had a snippet leading to the EN unsubscribe page - now fixed
- [x] For JA, BN, ID, NL, TR, ZH, update links in "general_opt_in_privacy_notice_general" lang string to new page names
- [x] [Final peace of mind QA pass by someone with fresh eyes](https://docs.google.com/spreadsheets/d/1LNa5Jnani1g_1YmJDFCHzVj2esJND9NadmPxQL3Z3Ss/edit#gid=124764889) 

For future followup in other tickets:
- [ ] I've added the `?mailing_list_id=62` tag to the Fossil Free News wrappers only - this could be added to other wrappers/mailing lists. I noticed though that if a user is not subscribed to list 62 but they click a link with this param, and they're only signed up to one list, the unsubscribe will not register, eg: https://act.350.org/unsubscribe/unsubscribe/?mailing_list_id=62&t=6&ak_proof=1&akid=%2E3307926%2EXlbU9d -- this edge case should be resolved if poss before we apply the mailing list params to all wrappers
- [ ] Thanks page logic / treatment of FF resubscribes
- [ ] Options for folks who want to receive fewer mailings
- [ ] Data tracking & dashboards

**Comment by rachelhbd** [comment]
Hey @sukhada everything is looking good for the translated versions except for one part -- the page titles -- since the position of the user name and the commas varies depending on the language.

Since using the dynamic "user name" snippets hasn't been working, it was previously hardcoded into the template after the page_title_custom_html custom page field, but looking at the translated variants this isn't going to work

I reached out to AK (email thread "Dynamic user names in page titles") about this and they recommended using the {% include_tmpl %} tag, which tells the templates to evaluate snippets:

```
{% if page.custom_fields.page_title_custom_html %}
		{% include_tmpl page.custom_fields.page_title_custom_html %}
	{% else %}
		<span class="{{ page.custom_fields.page_title_style }}">{{ page.title }}</span>
	{% endif %}
```

Would this be something we could add? Happy to break it into a separate issue if you want.

Also noting this is something it'd be good to look at in wrapper.html for all pages as Shannon suggests, but I can include that in the Sept sprint assuming that's more complex

**Comment by sukhada** [comment]
@rachelhbd Added a PR here: https://github.com/350org/ak_intl_v3/pull/372!

**Comment by rachelhbd** [comment]
thanks! and for taking care of the other page types there too :)

**Comment by rachelhbd** [comment]
hey @sukhada I did a bit more thinking about the final release process when this one is sorted and updated the steps above. Does this one need to go into Develop first before going live because [fa-nov-sprint-unsubscribe-page](https://github.com/350org/ak_intl_v3/tree/fa-nov-sprint-unsubscribe-page) branches off Develop?

**Comment by sukhada** [comment]
@rachelhbd yes it does!

**Comment by rachelhbd** [comment]
Alright thanks! I would like to go live with a set of multilingual pages using a staging template of some sort (just for a day or two before merging into v3, whilst I'm swapping out the old pages that won't work fully with the new templateset). I was thinking that could be the Unsubscribe templateset once it's been fully QA'd so it won't block/be blocked by anything, but do you anticipate the merge into Develop being a complex one? If so we could wait til it's merged in there and do our last full round of QA then

**Comment by sukhada** [comment]
@rachelhbd It is a bit of a tricky merge now that I'm looking at it - might be best to QA when it's merged into Develop!

**Comment by rachelhbd** [comment]
hey @sukhada I've [done a round of QA](https://docs.google.com/spreadsheets/d/1LNa5Jnani1g_1YmJDFCHzVj2esJND9NadmPxQL3Z3Ss/edit#gid=1703017732). I didn't find any browser/device issues, but did find:
-  **a bug with the "unsubscribe from all" link** - perhaps a JS issue of some sort?
This feature does work in the [original FA version](https://act.350.org/act/unsubscribe-new/?akid=.5148234.PNjCJz&template_set=350.org%20-%20International%20v3%20(FA%20sprint%20nov%2021)) but not in Develop or the templateset using the new wtr-fa-unsubscribe-page branch
Also I was wondering, if the link does rely on javascript rather than html, do you think having an unsub submit that relies on JS is robust enough here? This is a link we should be sure should works for all users. We could consider reviewing this as part of the Sept sprint - maybe we can have this as a use case for an automated test?

- **survey responses are being recorded as a sentence** - sorry I didn't spot that in FA's work before, it would be better for the values to be simplified so it's clear they include multilingual responses and allows us to tweak phrasing (which we've already had requests for)
This one is non-blocking and we could update in Sept, but might be a relatively quick fix and would keep our data in a better shape
The current parent field name is good (survey_unsubscribe_reason), for the values we could use the names from the language strings? eg. "frequency" "consent" etc? Here is how they match up:

> frequency=You send me too many emails
> consent=I never signed up for this mailing list
> difemail=I signed up with a different email
> campaigns=I'm no longer interested in 350.org's campaigns
> other=Other

**Comment by cjmabry** [comment]
> - a bug with the "unsubscribe from all" link - perhaps a JS issue of some sort?

@sukhada @rachelhbd Taking a look to see if I can find what is causing this.

**Comment by cjmabry** [comment]
> Also I was wondering, if the link does rely on javascript rather than html, do you think having an unsub submit that relies on JS is robust enough here? This is a link we should be sure should works for all users. We could consider reviewing this as part of the Sept sprint - maybe we can have this as a use case for an automated test?

@rachelhbd the way this works is that when a user presses **Unsubscribe from All**, Javascript checks the boxes for all the lists on the page, then submits the Unsubscribe. It does not appear there is an easy way to unsubscribe people from multiple lists at once via just a link, so I think this solution is optimal, although it might be worth reaching out to Action Kit to confirm.


Here is line for reference:
https://github.com/350org/ak_intl_v3/blob/e81a11d5df1feea58ec60c965036f9d1523b2d1d/unsubscribe.html#L492

That being said, I am unsure why this isn't firing here but was in the FA branch, as those unsubscribe.html pages are identical as far as I can tell.

**Comment by rachelhbd** [comment]
Thanks @cjmabry, confirming that @sukhada was able to fix this and it's now passing QA (see QA round 2): https://docs.google.com/spreadsheets/d/1LNa5Jnani1g_1YmJDFCHzVj2esJND9NadmPxQL3Z3Ss/edit#gid=2135428603

**Comment by rachelhbd** [comment]
**Email wrappers**

Leaving a note here that I did find some issues with QA where some of our wrappers relied on AK default settings. I've updated the default page and been through the individual wrappers, but noting they could do with another round of tidying up.

----
My request to AK:

> Hey AK, we're in the process of updating our unsubscribe pages and I want to check we're set up correctly. The new pages are collated in a multilingual campaign:
> 
> https://act.350.org/admin/core/multilingualcampaign/?q=unsubscribe&o=&ot=asc
> 
> In some cases, I've had to swap out the links, renaming the old page from "unsubscribe" to "unsubscribe-old" and the new one from "unsubscribe-new" to "unsubscribe"
> 
> I've seen that some of our older wrappers like "English - Global" is an old wrapper refer to "{% include "unsubscribe.html" %} or unsubscribe.txt in place of the unsubscribe links, and others had a {{ unsub_page_url }} snippet in place of the url for the plain text version.
> 
> Do you know where these files/snippets can be edited, and if that's still supported as a recommended approach?
> 
> Also, I'm doing as much QA as I can, but I know you occasionally picked up issues with an unsubscribe page or link in the past, so if you have an automated scanner to check clients' setups, please run this for us!

Their response:

> Hi Rachel,
> 
> The specific snippets that you mention:
> 
> {% include "unsubscribe.txt" %}
> {% include "unsubscribe.html" %}
> 
> are hardcoded defaults that live in the ActionKit codebase, so they aren't editable. The Unsubscribe Text and Unsubscribe HTML sections of the Email wrapper are there to extend or replace these -- here's the entire contents of each one since you may want to replace these with a different unsub page it sounds:
> 
> unsubscribe.txt
> ------------------------------
> You can unsubscribe from this mailing list at any time:
> http://{{ hostname }}/cms/unsubscribe/{{ unsub_page_name }}/
> 
> unsubscribe.html
> ------------------------------
> <p style="font-size:small"> <br> <a href="http://{{ hostname }}/cms/unsubscribe/{{ unsub_page_name }}/">You can unsubscribe from this mailing list at any time.</a> </p>
> 
> The only interesting bit there is the {{ unsub_page_name }} variable. This pulls up an Unsubscribe page name per this precedence order:
> 
> 1. The Unsubscribe page that has "Use In Mail Wrapper" marked (see under Unsubscribe Options on the unsub page creation screen), if one exists. We ensure that no more than one page can have this setting at the same time. Right now this page is "unsubscribe" id 15862
> 
> 2. If no page has "Use In Mail Wrapper" checked, then we insert the name of the Unsubscribe page with the lowest id, which is the preloaded one:
> 
> https://act.350.org/admin/core/unsubscribepage/5/change/
> 
> 350 has changed the name to "unsubscribe-old" but the id will always be the lowest, so it could be dangerous to hide that page and make it inaccessible.
> 
> It would be totally fine to replace all mentions of {% include "unsubscribe.txt" %} or {% include "unsubscribe.html" %} with their contents above and then edit as desired. That's most likely what you want to do because you'll want each of the language specific wrappers to link to a page in the same language.
